
<launch>

  <!-- record data to RTAB-Map database format (like a ROS bag, but usable in RTAB-Map for Windows/Mac OS X) -->
  <!-- This demo works with demo_mapping.bag -->

  <param name="use_sim_time" type="bool" value="True"/>

  <node name="data_recorder" pkg="rtabmap" type="data_recorder" output="screen">
	<param name="output_file_name" value="output.db" type="string"/>  
	
	<param name="frame_id" type="string" value="base_footprint"/>
	
	<param name="subscribe_odometry" type="bool" value="true"/>
	<param name="subscribe_depth" type="bool" value="true"/>
	<param name="subscribe_laserScan" type="bool" value="true"/>
	
	<remap from="odom" to="/az3/base_controller/odom"/>
	<remap from="scan" to="/jn0/base_scan"/>

	<remap from="rgb/image" to="data_throttled_image"/>
  	<remap from="depth/image" to="data_throttled_image_depth"/>
  	<remap from="rgb/camera_info" to="data_throttled_camera_info"/>
  	
  	<param name="rgb/image_transport" type="string" value="compressed"/>
    <param name="depth/image_transport" type="string" value="compressedDepth"/>
	
	<param name="queue_size" type="int" value="10"/>
  </node>

</launch>
